### PR TITLE
#1009 integration with JPA EntityManager - EntityListeners

### PR DESCRIPTION
**Pull request by IlyaNerd**
#1009

### CODE REVIEW

**Comment by bartoszwalacik** [comment]
great, I need a while for code review

File: javers-spring-jpa/aop.xml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE aspectj PUBLIC "-//AspectJ//DTD//EN" "http://www.eclipse.org/aspectj/dtd/aspectj.dtd">
```
Comment (bartoszwalacik) [inline javers-spring-jpa/aop.xml:1]: Can we do this without xml? I mean in Java Config

File: javers-spring-jpa/src/main/java/org/javers/spring/jpa/JaversEntityListener.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+import javax.persistence.PostRemove;
+import javax.persistence.PostUpdate;
+import java.util.Map;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * JPA entity listener to commit auditing changes. To get this one flying be
+ * sure you add @EnableJaversEntityListeners to your application config
+ * and add this entity listener in your {@code orm.xml} as follows:
```
Comment (bartoszwalacik) [inline javers-spring-jpa/src/main/java/org/javers/spring/jpa/JaversEntityListener.java:21]: again, xml is passé, is there another option to kick it of without xml?

Comment (IlyaNerd) [reply javers-spring-jpa/src/main/java/org/javers/spring/jpa/JaversEntityListener.java:21]: thats just a documentation how user can trigger this listener, if they like/need xml there is an option, which hibernate provides, not javers
and there are two options specified - xml or @EntityListeners, whichever user need
mb we should clarify this piece of javadoc if its confusing..

File: javers-spring-jpa/aop.xml (hunk 1 above)
Comment (IlyaNerd) [reply javers-spring-jpa/aop.xml:1]: so there is no such thing as java config at this point, its instruction for aspectj compiler (ajc) what to weave at compile time.
the link you gave has one diff with our case - we need to inject 3 beans somehow into a listener which is not created by spring, but by a EntityManager at startup. If you look at this class in build directory, you will see that in default constructor there is additional code generated by aspectJ at compile time. For that we are using spring feature - @Configured with aspectJ - https://www.baeldung.com/spring-inject-bean-into-unmanaged-objects . Same thing does spring-data-jpa with their AuditingEntityListener.
There is also another option to autowire into such objects - special class with ApplicationContext static field or having this listener with static fields and resolving them later, which in my opinion is kinda ugly. - https://stackoverflow.com/a/60651528
We could remove the xml, but then ajc will weave entire module and will find 3 classes to weave (this one, transactional decorator and javers spring data aspect), not sure if its desirable to alter their logic.

**Pull request closed without merging**